### ISSUE TITLE
Empty output while converting `https://graph.microsoft.com/beta/$metadata`

### ISSUE THREAD

**Issue by imWildCat** [oasis-tcs/odata-openapi#248]
Hello! Thanks for building such a great project!

Currently, I cannot convert the graph API odata metadata to JSON:

```
$ node_modules/odata-openapi/lib/cli.js ./ms-api.xml
```

```json
{"openapi":"3.0.2","info":{"title":"OData CSDL document","description":"","version":""},"paths":{},"components":{"schemas":{}}}
```

`ms-api.openapi3.json (END)`

**Comment by ralfhandl** [comment]
Hi @imWildCat,

Could you try running the tool via 

~~~sh
$ npx odata-openapi3 ./ms-api.xml
~~~

This works fine on my machine with the latest ms-graph-v1.0.edmx:

~~~pwsh
Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---          13/06/2023    17:30        2181042 ms-graph-v1.0.edmx
-a---          13/06/2023    17:30      148161187 ms-graph-v1.0.openapi3.json
~~~

Thanks in advance!

**Comment by imWildCat** [comment]
Thanks for the reply, @ralfhandl !

```
➜  Edge_Downloads npx odata-openapi3 ./metadata.xml
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/odata-openapi3 - Not found
npm ERR! 404
npm ERR! 404  'odata-openapi3@*' is not in this registry.
npm ERR! 404
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! A complete log of this run can be found in: /Users/[user]/.npm/_logs/2023-06-13T17_38_53_009Z-debug-0.log
➜  Edge_Downloads npm install -g odata-openapi


added 6 packages in 569ms

1 package is looking for funding
  run `npm fund` for details
➜  Edge_Downloads npx odata-openapi3 ./metadata.xml
./metadata.openapi3.json
/Users/[user]/.nvm/versions/node/v16.17.1/lib/node_modules/odata-openapi/lib/edm.js:120
        element = element.find(
                          ^

TypeError: Cannot read properties of undefined (reading 'find')
    at module.exports.EDM.#processAnnotations (/Users/[user]/.nvm/versions/node/v16.17.1/lib/node_modules/odata-openapi/lib/edm.js:120:27)
    at module.exports.EDM.#processSchemas (/Users/[user]/.nvm/versions/node/v16.17.1/lib/node_modules/odata-openapi/lib/edm.js:101:31)
    at module.exports.EDM.addDocument (/Users/[user]/.nvm/versions/node/v16.17.1/lib/node_modules/odata-openapi/lib/edm.js:32:25)
    at Object.module.exports.csdl2openapi (/Users/[user]/.nvm/versions/node/v16.17.1/lib/node_modules/odata-openapi/lib/csdl2openapi.js:69:9)
    at convert (/Users/[user]/.nvm/versions/node/v16.17.1/lib/node_modules/odata-openapi/lib/cli.js:89:23)
    at Object.<anonymous> (/Users/[user]/.nvm/versions/node/v16.17.1/lib/node_modules/odata-openapi/lib/cli.js:68:3)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Module.load (node:internal/modules/cjs/loader:1004:32)
    at Function.Module._load (node:internal/modules/cjs/loader:839:12)
➜  Edge_Downloads
```

**Comment by ralfhandl** [comment]
@imWildCat please try in a package where you've already installed `odata-openapi`, for example
~~~sh
mkdir foo
cd foo
npm init -yes
npm install odata-openapi
npx odata-openapi3 -h
~~~

That should list the usage options.

**Comment by imWildCat** [comment]
Thanks!

Still similar issue:

```
➜  f npm init -yes
npm install odata-openapi
npx odata-openapi3 -h
Wrote to /Users/[user]/Downloads/Edge_Downloads/f/package.json:

{
  "name": "f",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC"
}



added 6 packages, and audited 7 packages in 520ms

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities
Usage: odata-openapi3 <options> <source file>
Options:
 --basePath              base path (default: /service-root)
 -d, --diagram           include YUML diagram
 -h, --help              show this info
 --host                  host (default: localhost)
 -o, --openapi-version   3.0.0 to 3.0.3 or 3.1.0 (default: 3.0.2)
 -p, --pretty            pretty-print JSON result
 --scheme                scheme (default: http)
 -t, --target            target file (default: source file basename + .openapi3.json)
 --skipBatchPath         skips the generation of the $batch path, (default: false)
 --title                 default title if none is annotated
 --description           default description if none is annotated
➜  f ls
metadata.xml      node_modules      package-lock.json package.json
➜  f npx odata-openapi3 ./metadata.xml
➜  f cat package.json
{
  "name": "f",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "odata-openapi": "^0.23.0"
  }
}
➜  f npx odata-openapi3 ./metadata.xml
./metadata.openapi3.json
/Users/[user]/Downloads/Edge_Downloads/f/node_modules/odata-openapi/lib/edm.js:120
        element = element.find(
                          ^

TypeError: Cannot read properties of undefined (reading 'find')
    at module.exports.EDM.#processAnnotations (/Users/[user]/Downloads/Edge_Downloads/f/node_modules/odata-openapi/lib/edm.js:120:27)
    at module.exports.EDM.#processSchemas (/Users/[user]/Downloads/Edge_Downloads/f/node_modules/odata-openapi/lib/edm.js:101:31)
    at module.exports.EDM.addDocument (/Users/[user]/Downloads/Edge_Downloads/f/node_modules/odata-openapi/lib/edm.js:32:25)
    at Object.module.exports.csdl2openapi (/Users/[user]/Downloads/Edge_Downloads/f/node_modules/odata-openapi/lib/csdl2openapi.js:69:9)
    at convert (/Users/[user]/Downloads/Edge_Downloads/f/node_modules/odata-openapi/lib/cli.js:89:23)
    at Object.<anonymous> (/Users/[user]/Downloads/Edge_Downloads/f/node_modules/odata-openapi/lib/cli.js:68:3)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Module.load (node:internal/modules/cjs/loader:1004:32)
    at Function.Module._load (node:internal/modules/cjs/loader:839:12)
➜  f
```

I also tried node.js 18:

```
➜  f nvm use 18
Now using node v18.10.0 (npm v8.19.2)
➜  f npx odata-openapi3 ./metadata.xml
./metadata.openapi3.json
/Users/[user]/Downloads/Edge_Downloads/f/node_modules/odata-openapi/lib/edm.js:120
        element = element.find(
                          ^

TypeError: Cannot read properties of undefined (reading 'find')
    at #processAnnotations (/Users/[user]/Downloads/Edge_Downloads/f/node_modules/odata-openapi/lib/edm.js:120:27)
    at #processSchemas (/Users/[user]/Downloads/Edge_Downloads/f/node_modules/odata-openapi/lib/edm.js:101:31)
    at module.exports.EDM.addDocument (/Users/[user]/Downloads/Edge_Downloads/f/node_modules/odata-openapi/lib/edm.js:32:25)
    at module.exports.csdl2openapi (/Users/[user]/Downloads/Edge_Downloads/f/node_modules/odata-openapi/lib/csdl2openapi.js:69:9)
    at convert (/Users/[user]/Downloads/Edge_Downloads/f/node_modules/odata-openapi/lib/cli.js:89:23)
    at Object.<anonymous> (/Users/[user]/Downloads/Edge_Downloads/f/node_modules/odata-openapi/lib/cli.js:68:3)
    at Module._compile (node:internal/modules/cjs/loader:1149:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1203:10)
    at Module.load (node:internal/modules/cjs/loader:1027:32)
    at Module._load (node:internal/modules/cjs/loader:868:12)

Node.js v18.10.0
```
[metadata.xml.zip](https://github.com/oasis-tcs/odata-openapi/files/11748568/metadata.xml.zip)

**Comment by ralfhandl** [comment]
@imWildCat 

Bug is fixed with `odata-openapi@0.23.1`.

Thanks for finding this!

**Comment by imWildCat** [comment]
@ralfhandl thank you so much!
